### DISCUSSION TITLE
Airflow Helm Chart how to install providers

### DISCUSSION THREAD

**Discussion by ossamasgr** [apache/airflow#26927] [Q&A] [answered]
HI 
i have installed bitnami/airflow chart and i want to connect it to a spark cluster 
![image](https://user-images.githubusercontent.com/59144753/194430268-d0c8c74f-bcdf-447f-8cfa-ff59a8a38aa2.png)

i have installed this packages on the worker,web,scheduler 
using extraPipPackages : 
```
  extraPipPackages:
    - "apache-airflow-providers-apache-spark==2.2.0"
    - "apache-airflow-providers-celery"
    - "apache-airflow-providers-apache-spark"
    - "pyspark"
    - "apache-airflow"
```

and i also tried executing into the pods and installing the packages directly 
but nothing is working for me for the moment 

i have imported a dag file to test but it's not getting imported 

![image](https://user-images.githubusercontent.com/59144753/194430764-be852b2e-95e8-4a1d-aa80-ca94ddbfd67c.png)

couldn't find anything on the documentation 

i hope someone can guide me to the right direction

**Comment by potiuk** [comment, marked as answer]
It's not chart which the community manages here - you should likely ask in Bitnami's repo. 

We actually consider any approach where "extra pip packages" are used in similar way as a very, very bad and dangerous anti-pattern (useful possibly only when you iterate on it and debug stuff). It is precisely the reason why you have a lot of problems like that . You can read more here: https://airflow.apache.org/docs/docker-stack/entrypoint.html#installing-additional-requirements.

Instead of getting it fixed, I strongly recommend you to build your own image where you add the providers https://airflow.apache.org/docs/docker-stack/build.html#adding-new-pypi-packages-individually and switching to the Official Helm chart https://airflow.apache.org/docs/helm-chart/stable/index.html rather than the bitnami chart - we are maintaining chart in the same way we maintain/release airflow.

**Reply by ossamasgr** [reply]
Thank you so much for your respond that was really helpful
